### DISCUSSION TITLE
3-phase breakers in the app

### DISCUSSION THREAD

**Discussion by mixermek** [MarkBryanMilligan/LanternPowerMonitor#40] [Q&A]
Will there be any eventually? I saw check ins with "3-phase logic" in the comment.
Here in Sweden there is quite common with 3-phase breakers and it would simplify things.

A side note. If You want......after going for c/c++ in the app... I could try to "Europifie" the app with layout for Scandinavian and some other european contries. see link for example. https://img.tradera.net/images/238/373681238_d7b903ef-2619-4c9a-88e7-23d29d47470e.jpg

(Realized after posting this should NOT be under Q&A. Pls move or delete)

**Comment by hdoedens** [comment]
I am looking into this project as well and are also very interested in the European equivalent to this. I am currently trying to get small set-up working on a breadboard, just to see if I can get it to work at all.
Here in The Netherlands 3-phase is very common as well, but maybe more imported: the minimum voltage is 230V instead of 120V.

**Comment by MarkBryanMilligan** [comment]
The android app already allows configuration of 3-phase 230V systems.  The hubs, on first startup, auto-calibrate to 230V if they detect 50Hz.  The phase configuration is done in the panel config on the android app.  I haven't had a chance to add that config to the iOS app yet.

**Reply by mixermek** [reply]
Ah good. I've just got delivery of the components needed to build the interface board so I haven't had the opportunity to test the app with detected 230V 50Hz yet (Android user) actually planning to do a test setup this weekend :-D

**Reply by hdoedens** [reply]
I don't have the Android app. I could borrow a phone to see how the phase config is reflected in the breaker config. I can then get the config from the server and add the rest manually (I already modified the code to be able to read the config from a local file).
In the config, I see the `phase_offset_ns` property. Is it possible to calculate the offset manually?

I would like to know if it is possible to feed in the actual voltages another wat. I have the live voltage available by using DSMR reader and reading the smart meter. This is posted to a MQTT topic. I already setup some java code to 'inject' the Voltage into the calculation. But this is where I got stuck. The calculation is quite complex and I don't see where I could swap the measured Voltage with the 'MQTT Voltage'.

Would this be possible at all?
